src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
```

Comment:
**Optimization Issue**: Removing the loop that adjusts the cache size during a capacity change may lead to excessive memory usage if the capacity is reduced significantly. <br> **Fix**: Reintroduce the loop to evict entries when the new capacity is less than the current size of the cache. This ensures that the cache does not hold more entries than the specified capacity. <br> **Code Suggestion**: 
 ```
 +        for (int i = data.size(); i > newCapacity; i--) {
 +            Entry<K, V> evicted = evict();
 +            data.remove(evicted.getKey());
 +        }
 ```